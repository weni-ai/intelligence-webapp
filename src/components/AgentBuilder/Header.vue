<template>
  <header
    :class="[
      'agent-builder-header',
      {
        'agent-builder-header--actions-lg': actionsSize === 'lg',
        'agent-builder-header--actions-none': actionsSize === 'none',
      },
      props.class,
    ]"
    data-testid="agent-builder-header"
  >
    <section class="agent-builder-header__title">
      <UnnnicIntelligenceText
        tag="h1"
        size="title-sm"
        family="secondary"
        weight="bold"
        color="neutral-darkest"
        data-testid="agent-builder-header-title"
      >
        {{ currentView?.title }}
      </UnnnicIntelligenceText>
      <UnnnicIntelligenceText
        tag="h2"
        size="body-gt"
        family="secondary"
        color="neutral-cloudy"
        data-testid="agent-builder-header-description"
      >
        {{ currentView?.description }}

        <SupervisorHeaderDetails
          v-if="
            currentView.page === 'supervisor' &&
            featureFlagsStore.flags.newSupervisor
          "
        />
      </UnnnicIntelligenceText>
    </section>

    <slot name="actions" />
  </header>

  <UnnnicDivider
    v-if="withDivider"
    ySpacing="md"
    data-testid="agent-builder-header-divider"
  />
</template>

<script setup>
import { computed } from 'vue';
import { useRoute } from 'vue-router';

import useAgentBuilderViews from '@/composables/useAgentBuilderViews';

import SupervisorHeaderDetails from './Supervisor/SupervisorHeaderDetails.vue';

import { useFeatureFlagsStore } from '@/store/FeatureFlags';

const route = useRoute();

const props = defineProps({
  class: {
    type: String,
    default: '',
  },

  withDivider: {
    type: Boolean,
    default: true,
  },

  actionsSize: {
    type: String,
    default: 'md',
    validator: (value) => {
      return ['none', 'md', 'lg'].includes(value);
    },
  },
});

const featureFlagsStore = useFeatureFlagsStore();

const currentView = computed(() => {
  const views = useAgentBuilderViews();
  return views.value.find((e) => e.page === route.name) || views.value[0];
});
</script>

<style lang="scss" scoped>
.agent-builder-header {
  display: grid;
  grid-template-columns: 9fr 3fr;
  gap: $unnnic-spacing-sm;
  align-items: center;

  &__title {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-xs;
  }

  &--actions-lg {
    grid-template-columns: 6fr 6fr;
  }

  &--actions-none {
    grid-template-columns: 9fr 0fr;
  }
}
</style>
