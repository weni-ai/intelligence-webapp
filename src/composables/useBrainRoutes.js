import { computed } from 'vue';
import i18n from '@/utils/plugins/i18n';
import { useFeatureFlagsStore } from '@/store/FeatureFlags';

export default function useBrainRoutes() {
  const t = (value) => i18n.global.t(value);

  const isAgentsTeamEnabled = useFeatureFlagsStore().flags.agentsTeam;

  return computed(() =>
    [
      {
        title: 'monitoring',
        description: t('router.monitoring.description'),
        page: 'router-monitoring',
        icon: 'bar_chart',
        preview: false,
        tag: t('new'),
      },
      {
        title: 'personalization',
        page: 'router-personalization',
        icon: 'person',
        preview: true,
      },
      isAgentsTeamEnabled
        ? {
            title: 'agents-team',
            description: t('router.agents_team.description'),
            page: 'router-agents-team',
            icon: 'workspaces',
            preview: false,
          }
        : null,
      {
        title: 'content',
        description: t('content_bases.description'),
        page: 'router-content',
        icon: 'article',
        preview: true,
      },
      isAgentsTeamEnabled
        ? null
        : {
            title: 'actions',
            description: t('router.actions.description'),
            page: 'router-actions',
            icon: 'bolt',
            preview: true,
          },
      {
        title: 'tunings',
        page: 'router-tunings',
        icon: 'settings',
        preview: true,
      },
    ].filter((obj) => obj),
  );
}
