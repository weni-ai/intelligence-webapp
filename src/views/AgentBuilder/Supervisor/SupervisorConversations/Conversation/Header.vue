<template>
  <header class="conversation__header">
    <section class="header__info">
      <SupervisorUsername
        :username="conversation?.username"
        size="title-sm"
      />

      <UnnnicIntelligenceText
        tag="h2"
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        weight="regular"
        data-testid="conversation-title"
      >
        {{ formattedUrn }}
      </UnnnicIntelligenceText>

      <section
        v-if="topics.length > 0"
        class="header__topics"
      >
        <UnnnicTag
          v-for="topic in topics"
          :key="topic"
          scheme="neutral-light"
          :text="topic"
        />
      </section>
    </section>

    <button
      class="header__close-button"
      data-testid="close-button"
      @click="supervisorStore.selectConversation(null)"
    >
      <UnnnicIcon
        data-testid="close-button-icon"
        icon="close"
        size="md"
        scheme="neutral-cloudy"
      />
    </button>
  </header>
</template>

<script setup>
import { computed } from 'vue';

import { useSupervisorStore } from '@/store/Supervisor';
import i18n from '@/utils/plugins/i18n';
import { formatWhatsappUrn } from '@/utils/formatters';
import SupervisorUsername from '@/components/AgentBuilder/Supervisor/SupervisorUsername.vue';

const supervisorStore = useSupervisorStore();

const conversation = computed(() => supervisorStore.selectedConversation);
const formattedUrn = computed(() =>
  i18n.global.t('agent_builder.supervisor.contact_urn', {
    urn: formatWhatsappUrn(conversation.value?.urn),
  }),
);
const topics = computed(
  () => {
    if (typeof conversation.value?.topics === 'string') {
      return [conversation.value?.topics];
    }

    if (conversation.value?.topics?.length === 0) {
      return conversation.value?.topics;
    }

    return [];
  },
);
</script>

<style lang="scss" scoped>
.conversation__header {
  padding: $unnnic-spacing-sm;

  display: flex;
  justify-content: space-between;

  border-bottom: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  .header__info {
    overflow: hidden;

    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-xs;
  }

  .header__topics {
    display: flex;
    gap: $unnnic-spacing-xs;
    flex-wrap: wrap;

    :deep(.unnnic-tag) {
      background-color: $unnnic-color-neutral-light;
    }

    :deep(.unnnic-tag__label) {
      color: $unnnic-color-neutral-cloudy;
    }
  }

  .header__close-button {
    background-color: transparent;
    border: none;

    display: flex;

    cursor: pointer;
  }
}
</style>
