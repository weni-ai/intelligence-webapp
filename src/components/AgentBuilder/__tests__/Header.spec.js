import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach, vi } from 'vitest';

import Header from '../Header.vue';
import useBrainRoutes from '@/composables/useBrainRoutes';

vi.mock('@/composables/useBrainRoutes', () => {
  return {
    default: vi.fn(() => {
      return {
        value: [
          {
            title: 'Test Route',
            description: 'Test Description',
            page: 'test-page',
            icon: 'test-icon',
          },
          {
            title: 'Test Route 2',
            description: 'Test Description 2',
            page: 'test-page-2',
            icon: 'test-icon-2',
          },
        ],
      };
    }),
  };
});

vi.mock('vue-router', () => {
  return {
    useRoute: vi.fn(() => ({
      name: 'test-page',
    })),
  };
});

describe('Header.vue', () => {
  let wrapper;

  const title = () =>
    wrapper.findComponent('[data-testid="agent-builder-header-title"]');
  const description = () =>
    wrapper.findComponent('[data-testid="agent-builder-header-description"]');
  const divider = () =>
    wrapper.findComponent('[data-testid="agent-builder-header-divider"]');
  const header = () => wrapper.find('[data-testid="agent-builder-header"]');

  const createWrapper = (props = {}) => {
    return shallowMount(Header, {
      props: {
        withDivider: true,
        actionsSize: 'md',
        ...props,
      },
      global: {
        stubs: {
          UnnnicIntelligenceText: false,
        },
      },
    });
  };

  beforeEach(() => {
    wrapper = createWrapper();
  });

  describe('Rendering', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders correctly with default props', () => {
      expect(header().exists()).toBe(true);
      expect(title().exists()).toBe(true);
      expect(description().exists()).toBe(true);
      expect(divider().exists()).toBe(true);
    });

    it('renders the correct title and description from brainRoutes', () => {
      expect(title().text()).toBe('Test Route');
      expect(description().text()).toBe('Test Description');
    });

    it('does not render divider when withDivider is false', async () => {
      await wrapper.setProps({ withDivider: false });
      expect(divider().exists()).toBe(false);
    });

    it('applies the correct class when actionsSize is lg', async () => {
      expect(header().classes()).not.toContain(
        'agent-builder-header--actions-lg',
      );

      await wrapper.setProps({ actionsSize: 'lg' });
      expect(header().classes()).toContain('agent-builder-header--actions-lg');
    });
  });

  describe('Props validation', () => {
    it('accepts valid actionsSize values', () => {
      const validator = Header.props.actionsSize.validator;

      expect(validator('md')).toBe(true);
      expect(validator('lg')).toBe(true);
      expect(validator('invalid')).toBe(false);
    });
  });

  describe('Slot rendering', () => {
    it('renders content in the actions slot', () => {
      const slotContent = '<button>Test Button</button>';

      wrapper = shallowMount(Header, {
        slots: {
          actions: slotContent,
        },
        global: {
          stubs: {
            UnnnicIntelligenceText: true,
            UnnnicDivider: true,
          },
        },
      });

      expect(wrapper.html()).toContain(slotContent);
    });
  });
});
