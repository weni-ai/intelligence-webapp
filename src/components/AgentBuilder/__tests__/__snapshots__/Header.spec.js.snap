// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue > Rendering > should match snapshot 1`] = `
"<header data-v-e01a57d3="" class="agent-builder-header" data-testid="agent-builder-header">
  <section data-v-e01a57d3="" class="agent-builder-header__title">
    <h1 data-v-ec8da4ad="" data-v-e01a57d3="" class="unnnic-text unnnic-text--size-title-sm unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-bold unnnic-text--color-neutral-darkest" data-testid="agent-builder-header-title">Test Route</h1>
    <h2 data-v-ec8da4ad="" data-v-e01a57d3="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy" data-testid="agent-builder-header-description">Test Description</h2>
  </section>
</header>
<unnnic-divider-stub data-v-e01a57d3="" yspacing="md" scheme="neutral-soft" data-testid="agent-builder-header-divider"></unnnic-divider-stub>"
`;
